### PR TITLE
[CI] Move the linting jobs from CircleCI to Gitlab

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR moves all the tooling linting jobs from CircleCI to Gitlab.

### Motivation

We aim to get rid of the CircleCI jobs in the near future.

### Additional Notes

https://datadoghq.atlassian.net/browse/ACIX-403

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -126,69 +126,6 @@ jobs:
           no_output_timeout: 20m
           command: inv -e linter.go
```
Comment (chouetz) [inline .circleci/config.yml:127]: and why not this one as well? It seems like it's already launched in `.linux_lint` so we can probably deactivate it, wdyt?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 include:
   - .gitlab/.pre/cancel-prev-pipelines.yml
+  - .gitlab/.pre/linters.yml
```
Comment (chouetz) [inline .gitlab-ci.yml:4]: 💬 for a following PR
We might want to be more precise and split the linters in 2 categories, the "traditional" language linters (shell, python, go) from the additional ones (licenses, copyrights, gitlab (hidden in `test_gitlab_configuration`).
My suggestion:
- extract the lint jobs from `source_test` and create a `language_linters.yml` (and associated stage) with lint shell, go & python
- put all other linters in a `test_pr_configuration` that can stay in the `.pre` stage (which is supposed to be a "[before build](https://docs.gitlab.com/ee/ci/yaml/#stages)" stage)

wdyt?

**Review by chouetz — APPROVED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (amenasria) [reply .gitlab-ci.yml:4]: I like the before/after build split depending on what it lints, I won't do this in this PR though !

File: .circleci/config.yml (hunk 1 above)
Comment (amenasria) [reply .circleci/config.yml:127]: Good point

Comment (amenasria) [reply .circleci/config.yml:127]: But I chose to remove the circleCI jobs in a different PR, so I will however put the circleCI code back.

**Pull request merged**